PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/common/utils/WebSocketSSLProxy.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.toolkit.lib.common.utils;
+
+import com.neovisionaries.ws.client.*;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+public class WebSocketSSLProxy {
+    private static final Logger logger = Logger.getLogger(WebSocketSSLProxy.class.getName());
+    private static final int DEFAULT_BUFFER_SIZE = 8192;
+    private int bufferSize = DEFAULT_BUFFER_SIZE;
+    private int connectTimeout = 0;
+    private String webSocketServerUri;
+    private String id;
+    private String password;
+    private ServerSocket serverSocket;
+    private Thread thread;
+    private WebSocket webSocket;
+
+    public WebSocketSSLProxy(String webSocketServerUri, String id, String password) {
+        this.webSocketServerUri = webSocketServerUri;
+        this.id = id;
+        this.password = password;
+    }
+
+    public void start() throws IOException {
+        close();
+        // InetAddress.getByName(null) points to the loopback address (127.0.0.1)
+        serverSocket = new ServerSocket(0, 1, InetAddress.getByName(null));
+        thread = new Thread(() -> {
+            try {
+                for (;;) {
+                    Socket clientSocket = serverSocket.accept();
+                    createWebSocketToSocket(clientSocket);
+                    pipeSocketDataToWebSocket(clientSocket);
+                }
+            } catch (IOException | WebSocketException e) {
```

Comment:
it seems the thread will only be stopped when interrupted.
I think `InterruptedException` should be handled/swallowed here since it's intended.